stress/src/metrics_overflow.rs

```diff
@@ -0,0 +1,43 @@
+/*
+    Stress test results:
+    OS: Ubuntu 22.04.3 LTS (5.15.146.1-microsoft-standard-WSL2)
+    Hardware: AMD EPYC 7763 64-Core Processor - 2.44 GHz, 16vCPUs,
+    RAM: 64.0 GB
+    4.5M /sec
+*/
+
+use lazy_static::lazy_static;
+use opentelemetry::{
+    metrics::{Counter, MeterProvider as _},
+    KeyValue,
+};
+use opentelemetry_sdk::metrics::{ManualReader, SdkMeterProvider};
+use rand::{rngs::SmallRng, Rng, SeedableRng};
+use std::borrow::Cow;
+
+mod throughput;
+
+lazy_static! {
+    static ref PROVIDER: SdkMeterProvider = SdkMeterProvider::builder()
+        .with_reader(ManualReader::builder().build())
+        .build();
+    static ref ATTRIBUTE_VALUES: [&'static str; 10] = [
+        "value1", "value2", "value3", "value4", "value5", "value6", "value7", "value8", "value9",
+        "value10"
+    ];
+    static ref COUNTER: Counter<u64> = PROVIDER
+        .meter(<&str as Into<Cow<'static, str>>>::into("test"))
+        .u64_counter("hello")
+        .init();
+}
+
+fn main() {
+    for v in 0..2001 {
+        COUNTER.add(100, &[KeyValue::new("A", v.to_string())]);
+    }
+    throughput::test_throughput(test_counter);
+}
+
+fn test_counter() {
+    COUNTER.add(1, &[KeyValue::new("A", "2001")]);
```

Comment:
I think we should use random `ATTRIBUTE_VALUES` here to have a fair comparison with the regular stress test for metrics. Emitting the same dimension here would always give better results than this: https://github.com/open-telemetry/opentelemetry-rust/blob/aca4b52e20ee6728c91eb74744adfffa68e061d6/stress/src/metrics.rs#L2-L6

Reply:
The goal for this test was to make sure that in the event of some rogue input, the process won't crash due to metrics i.e are we keeping our memory under cap? are we doing excessive logging (like we were doing prior to this PR) ? etc.

I think this test can do the below to simulate that?
`COUNTER.add(1, &[KeyValue::new("A", i++)]);`

Will keep iterating on this.

Reply:
We could remove `ATTRIBUTE_VALUES` if we are not using them.